core/services/fluxmonitorv2/flux_monitor.go

```diff
@@ -655,25 +662,41 @@ func (fm *FluxMonitor) respondToNewRoundLog(log flux_aggregator_wrapper.FluxAggr
 	}
 
 	// Call the v2 pipeline to execute a new job run
-	runID, answer, err := fm.pipelineRun.Execute(metaDataForBridge)
+	run, results, err := fm.runner.ExecuteRun(context.Background(), fm.spec, pipeline.JSONSerializable{Val: metaDataForBridge}, *fm.logger)
 	if err != nil {
-		newRoundLogger.Errorf("unable to fetch median price: %v", err)
-
+		logger.Errorw(fmt.Sprintf("error executing new run for job ID %v name %v", fm.spec.JobID, fm.spec.JobName), "err", err)
+		return
+	}
+	result := results.FinalResult()
+	answer, err := utils.ToDecimal(result.Values[0])
+	if err != nil {
+		logger.Errorw(fmt.Sprintf("error executing new run for job ID %v name %v", fm.spec.JobID, fm.spec.JobName), "err", err)
 		return
 	}
 
-	if !fm.isValidSubmission(logger.Default.SugaredLogger, *answer) {
+	if !fm.isValidSubmission(logger.Default.SugaredLogger, answer) {
 		return
 	}
 
 	if roundState.PaymentAmount == nil {
 		newRoundLogger.Error("roundState.PaymentAmount shouldn't be nil")
 	}
 
-	err = fm.submitTransaction(runID, *answer, roundState.RoundId)
+	ctx, cancel := postgres.DefaultQueryCtx()
+	defer cancel()
+	err = postgres.GormTransaction(ctx, fm.db, func(tx *gorm.DB) error {
+		runID, err2 := fm.runner.InsertFinishedRun(tx, run, results, false)
+		if err2 != nil {
+			return err2
+		}
+		err2 = fm.submitTransaction(tx, runID, answer, roundState.RoundId)
```

Comment:
Might be nicer to rename this to "insertTransaction" or "createTransaction" otherwise as JB pointed out, it looks like an I/O operation